src/main/java/lesson9Text/TextAnalyzer.java

```diff
@@ -0,0 +1,23 @@
+package lesson9Text;
+
+public class TextAnalyzer {
+
+    public static String analyzeText(String text){
+        if (text.equals("")){
+            return "this is an empty string";
+        }
+        return String.format("length: %s\nis palindrome: %b", getTextLength(text), isPalindrome(text));
+    }
+
+    public static boolean isPalindrome(String text) {
+        text = text.replaceAll("[^a-zA-Z0-9]","").toLowerCase();
+        StringBuilder stringBuilder = new StringBuilder(text);
```

Comment:
Odwrócenie tekstu można by wynieść do osobnej metody. Jest to krok algorytmu i dobrze jakby był dobrze opisany (w postaci nazwy metody).